BUILD.gn

```diff
@@ -0,0 +1,495 @@
+    "runtime/common/xwalk_paths.cc",
+    "runtime/common/xwalk_paths.h",
+    "runtime/common/xwalk_resource_delegate.cc",
+    "runtime/common/xwalk_resource_delegate.h",
+    "runtime/common/xwalk_runtime_features.cc",
+    "runtime/common/xwalk_runtime_features.h",
+    "runtime/common/xwalk_switches.cc",
+    "runtime/common/xwalk_switches.h",
+    "runtime/common/xwalk_system_locale.cc",
+    "runtime/common/xwalk_system_locale.h",
+    "runtime/renderer/android/xwalk_permission_client.cc",
+    "runtime/renderer/android/xwalk_permission_client.h",
+    "runtime/renderer/android/xwalk_render_thread_observer.cc",
+    "runtime/renderer/android/xwalk_render_thread_observer.h",
+    "runtime/renderer/android/xwalk_render_view_ext.cc",
+    "runtime/renderer/android/xwalk_render_view_ext.h",
+    "runtime/renderer/isolated_file_system.cc",
+    "runtime/renderer/isolated_file_system.h",
+    "runtime/renderer/xwalk_content_renderer_client.cc",
+    "runtime/renderer/xwalk_content_renderer_client.h",
+    "runtime/renderer/xwalk_render_thread_observer_generic.cc",
+    "runtime/renderer/xwalk_render_thread_observer_generic.h",
+  ]
+  include_dirs = [ "$root_gen_dir/xwalk" ]
+  defines = [
+    "XWALK_VERSION=\"$xwalk_version\"",
+    "CHROME_VERSION=\"$chrome_version\"",
+  ]
+  deps = [
+    ":generate_upstream_blink_version",
+    "//base",
+    "//base:base_static",
+    "//base:i18n",
+    "//base/third_party/dynamic_annotations",
+    "//cc",
+    "//components/autofill/content/browser",
+    "//components/autofill/content/renderer",
+    "//components/autofill/core/browser",
+    "//components/cdm/renderer",
+    "//components/devtools_http_handler",
+    "//components/resources:components_resources",
+    "//components/strings",
+    "//components/url_formatter",
+    "//components/user_prefs",
+    "//components/visitedlink/browser",
+    "//components/visitedlink/renderer",
+    "//content",
+    "//content/public/app:both",
+    "//content/public/browser",
+    "//content/public/child",
+    "//content/public/common",
+    "//content/public/utility",
+    "//gin",
+    "//ipc",
+    "//media",
+    "//net",
+    "//net:net_resources",
+    "//skia",
+    "//storage/browser",
+    "//storage/common",
+    "//third_party/WebKit/public:blink",
+    "//third_party/boringssl",
+    "//ui/base",
+    "//ui/gl",
+    "//ui/shell_dialogs",
+    "//ui/snapshot",
+    "//url",
+    "//v8",
+    "//xwalk/application:xwalk_application_lib",
+    "//xwalk/experimental",
+    "//xwalk/extensions",
+    "//xwalk/resources:xwalk_pak",
+    "//xwalk/resources:xwalk_resources",
+    "//xwalk/resources:xwalk_resources_100_percent_pak",
+    "//xwalk/resources:xwalk_resources_200_percent_pak",
+    "//xwalk/resources:xwalk_resources_300_percent_pak",
+    "//xwalk/sysapps",
+  ]
+
+  if (is_linux) {
+    deps += [ "//xwalk/resources:xwalk_strings" ]
+  }
+
+  # TODO: Add WebNotification for Linux
+  # WebNotification needs libgtk2ui and libnotify.
```

Comment:
Yes, I plan to solve it in another PR next to this one.
The unsolved problem is in xwalk/build/system.gyp we use pkg-config on libnotify.  I tried to follow "src/config/linux/pkg_config.gni" to write a new one, but failed.
I'll continue investigate on this later.